working/0546-patterns/patterns-feature-specification.md

```diff
@@ -483,7 +483,7 @@ expression.
 ### Variable pattern
 
 ```
-variablePattern ::= ( 'var' | 'final' | type )? identifier
+variablePattern ::= ( 'var' | 'final' | 'final'? type )? identifier
```

Comment:
Oh, dear. I think when I merged the changes from https://github.com/dart-lang/language/pull/2493 into this PR (since that PR landed first), they showed up here in the review. But I guess that works since it means you can review those changes here now. :)

> Do we need to say, in prose somewhere, that you are not allowed to use the `final` in a declaration like `var (x, final int y) = ....` You can use the `int y`, but not the `final int y`.

That's covered in the pattern context section, which says:

> It is a compile-time error if a variable pattern in a declaration context is marked with `var` or `final`.

I reworded a bit to clarify. I think it works OK now.